File: codes/python/gpt/partial/task_1/Solution.py
def allocate_bed(patient, wards):
    patient_id = patient['id']
    for ward in wards:
        if (ward['target_condition'] == patient['condition'] and
            (ward['gender_restriction'] == 'No Restriction' or 
             ward['gender_restriction'] == 'No Gender Restriction' or 
             (ward['gender_restriction'] == 'Male Only' and patient['gender'] == 'Male') or
             (ward['gender_restriction'] == 'Female Only' and patient['gender'] == 'Female')) and
            ward['age_restriction'][0] <= patient['age'] <= ward['age_restriction'][1] and
            all(req in ward['special_requirements'] for req in patient['special_requirements'])):
            
            if ward['available_beds']:
                assigned_bed = min(ward['available_beds'])
                return {
                    "patient_id": patient_id,
                    "assigned_ward": ward['ward_name'],
                    "assigned_bed": assigned_bed,
                    "assigned_satellite_hospital": None
                }
            
            for i in range(1, ward['overflow_capacity'] + 1):
                overflow_bed_label = f"Overflow-{i}"
                if overflow_bed_label not in ward['available_beds']:
                    return {
                        "patient_id": patient_id,
                        "assigned_ward": ward['ward_name'],
                        "assigned_bed": overflow_bed_label,
                        "assigned_satellite_hospital": None
                    }
            
            if ward['satellite_hospitals']:
                return {
                    "patient_id": patient_id,
                    "assigned_ward": None,
                    "assigned_bed": None,
                    "assigned_satellite_hospital": ward['satellite_hospitals'][0]
                }
    
    return {
        "patient_id": patient_id,
        "assigned_ward": None,
        "assigned_bed": None,
        "assigned_satellite_hospital": None
    }

def main():
    import sys
    import json
    input_data = json.load(sys.stdin)
    result = allocate_bed(input_data['patient'], input_data['wards'])
    print(json.dumps(result))

if __name__ == "__main__":
    main()

File: codes/go/gpt/partial/task_1/Solution.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

func allocateBed(patient map[string]interface{}, wards []interface{}) map[string]interface{} {
	result := map[string]interface{}{
		"patient_id":                 patient["id"],
		"assigned_ward":             nil,
		"assigned_bed":              nil,
		"assigned_satellite_hospital": nil,
	}

	for _, w := range wards {
		ward := w.(map[string]interface{})
		if matchesWard(patient, ward) {
			if len(ward["available_beds"].([]interface{})) > 0 {
				// Assign regular bed
				result["assigned_ward"] = ward["ward_name"]
				result["assigned_bed"] = ward["available_beds"].([]interface{})[0]
				return result
			} else if ward["overflow_capacity"].(int) > 0 {
				// Assign overflow bed
				result["assigned_ward"] = ward["ward_name"]
				result["assigned_bed"] = fmt.Sprintf("Overflow-%d", 1)
				return result
			} else if len(ward["satellite_hospitals"].([]interface{})) > 0 {
				// Assign satellite hospital
				result["assigned_satellite_hospital"] = ward["satellite_hospitals"].([]interface{})[0]
				return result
			}
		}
	}
	return result
}

func matchesWard(patient map[string]interface{}, ward map[string]interface{}) bool {
	// Check condition match
	if patient["condition"] != ward["target_condition"] {
		return false
	}
	// Check gender match
	if ward["gender_restriction"] != "No Restriction" && ward["gender_restriction"] != "No Gender Restriction" {
		if patient["gender"] == "Male" && ward["gender_restriction"] != "Male Only" {
			return false
		}
		if patient["gender"] == "Female" && ward["gender_restriction"] != "Female Only" {
			return false
		}
	}
	// Check age restriction
	age := patient["age"].(int)
	ageRestriction := ward["age_restriction"].([]interface{})
	if age < ageRestriction[0].(int) || age > ageRestriction[1].(int) {
		return false
	}
	// Check special requirements
	patientRequirements := patient["special_requirements"].([]interface{})
	wardRequirements := ward["special_requirements"].([]interface{})
	for _, req := range patientRequirements {
		found := false
		for _, wReq := range wardRequirements {
			if req == wReq {
				found = true
				break
			}
		}
		if !found {
			return false
		}
	}
	return true
}

func main() {
	var input struct {
		Patient map[string]interface{} `json:"patient"`
		Wards   []interface{}          `json:"wards"`
	}

	json.NewDecoder(os.Stdin).Decode(&input)
	result := allocateBed(input.Patient, input.Wards)

	output, _ := json.Marshal(result)
	fmt.Println(string(output))
}